internal/table/table.go

```diff
@@ -0,0 +1,167 @@
+package table
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"strings"
+	"unicode/utf8"
+
+	"github.com/ActiveState/cli/internal/logging"
+	"golang.org/x/crypto/ssh/terminal"
+)
+
+const dash = "\u2500"
+
+type FormatFunc func(string, ...interface{}) string
+
+type Table struct {
+	rows   [][]string
+	header []string
+	// TODO: Make column width part of a row struct?
+	columnWidths []int
+	padding      int
+	// TODO: Add logic to deal with max column size
+	maxSize int
+
+	headerFormatter FormatFunc
+	headerFormat    string
+}
+
+type writeBuffer struct {
+	bytes.Buffer
+}
+
+func createBuffer() *writeBuffer {
+	return &writeBuffer{}
+}
+
+func (b *writeBuffer) Write(str string, count int) *writeBuffer {
+	for i := 0; i < count; i++ {
+		b.WriteString(str)
+	}
+	return b
+}
+
+func (b *writeBuffer) String() string {
+	return b.String()
+}
+
+func Create(data interface{}) *Table {
```

Comment:
Why is this taking an `interface{}` value? Converting types is the job of the plain outputter. This package should just be dealing with opinionated data.